File: app/src/main/java/org/ole/planet/myplanet/model/Download.java
package org.ole.planet.myplanet.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Download implements Parcelable {

    public static final Parcelable.Creator<Download> CREATOR = new Parcelable.Creator<Download>() {
        public Download createFromParcel(Parcel in) {
            return new Download(in);
        }

        public Download[] newArray(int size) {
            return new Download[size];
        }
    };
    private String fileName;
    private int progress;
    private int currentFileSize;
    private int totalFileSize;
    private boolean completeAll;
    private boolean failed;
    private String message;
    private String fileUrl;

    public Download() {

    }

    private Download(Parcel in) {
        fileName = in.readString();
        progress = in.readInt();
        currentFileSize = in.readInt();
        totalFileSize = in.readInt();
        completeAll = in.readByte() != 0;
        failed = in.readByte() != 0;
        message = in.readString();
        fileUrl = in.readString();
    }

    public String getFileUrl() {
        return fileUrl;
    }

    public void setFileUrl(String fileUrl) {
        this.fileUrl = fileUrl;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public boolean isCompleteAll() {
        return completeAll;
    }

    public void setCompleteAll(boolean completeAll) {
        this.completeAll = completeAll;
    }

    public int getProgress() {
        return progress;
    }

    public void setProgress(int progress) {
        this.progress = progress;
    }

    public int getCurrentFileSize() {
        return currentFileSize;
    }

    public void setCurrentFileSize(int currentFileSize) {
        this.currentFileSize = currentFileSize;
    }

    public int getTotalFileSize() {
        return totalFileSize;
    }

    public void setTotalFileSize(int totalFileSize) {
        this.totalFileSize = totalFileSize;
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public boolean isFailed() {
        return failed;
    }

    public void setFailed(boolean failed) {
        this.failed = failed;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {

        dest.writeString(fileName);
        dest.writeInt(progress);
        dest.writeInt(currentFileSize);
        dest.writeInt(totalFileSize);
        dest.writeByte((byte) (completeAll ? 1 : 0));
        dest.writeByte((byte) (failed ? 1 : 0));
        dest.writeString(message);
        dest.writeString(fileUrl);
    }
}

File: app/src/main/java/org/ole/planet/myplanet/model/Download.kt
package org.ole.planet.myplanet.model

import android.os.Parcel
import android.os.Parcelable

class Download() : Parcelable {
    var fileName: String? = null
    var progress: Int = 0
    var currentFileSize: Int = 0
    var totalFileSize: Int = 0
    var completeAll: Boolean = false
    var failed: Boolean = false
    var message: String? = null
    var fileUrl: String? = null

    constructor(parcel: Parcel) : this() {
        fileName = parcel.readString()
        progress = parcel.readInt()
        currentFileSize = parcel.readInt()
        totalFileSize = parcel.readInt()
        completeAll = parcel.readByte() != 0.toByte()
        failed = parcel.readByte() != 0.toByte()
        message = parcel.readString()
        fileUrl = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(fileName)
        parcel.writeInt(progress)
        parcel.writeInt(currentFileSize)
        parcel.writeInt(totalFileSize)
        parcel.writeByte(if (completeAll) 1 else 0)
        parcel.writeByte(if (failed) 1 else 0)
        parcel.writeString(message)
        parcel.writeString(fileUrl)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Download> {
        override fun createFromParcel(parcel: Parcel): Download {
            return Download(parcel)
        }

        override fun newArray(size: Int): Array<Download?> {
            return arrayOfNulls(size)
        }
    }
}
